yarn-project/types/src/tx.ts

```diff
@@ -197,11 +197,19 @@ export class Tx {
     // We hash the full signed tx request object (this is, the tx request along with the signature),
     // just like Ethereum does.
     if (tx.txRequest) {
-      hashes.push(new TxHash(keccak(tx.txRequest.toBuffer())));
+      hashes.push(new TxHash(computeTxHash(wasm, tx.txRequest).toBuffer()));
     }
 
     // Return a tx hash if we have only one, or hash them again if we have both
     if (hashes.length === 1) return hashes[0];
-    else return new TxHash(keccak(Buffer.concat(hashes.map(h => h.buffer))));
+    else return new TxHash(keccak224(Buffer.concat(hashes.map(h => h.buffer))));
+  }
```

Comment:
I think we need to change most of this function: if we got tx.data, then we return the first nullifier, if we got txRequest, we computeTxHash. And we should drop the "combine two hashes" thing, right?

Reply:
I wasn't really sure what the 'combine two hashes' case was solving so I left it in for now. Eventually I would expect this function to simply return the first nullifier.

Reply:
It's gone